src/lib/Scenes/Artwork/Components/ArtworkActions.tsx

```diff
@@ -1,49 +1,98 @@
-import { Box, Flex, Sans } from "@artsy/palette"
+import { Box, color, Flex, Sans } from "@artsy/palette"
 import { ArtworkActions_artwork } from "__generated__/ArtworkActions_artwork.graphql"
+import { ArtworkActionsSaveMutation } from "__generated__/ArtworkActionsSaveMutation.graphql"
+import { Pin } from "lib/Icons/Pin"
 import SearchIcon from "lib/Icons/SearchIcon"
+import { isNull } from "lodash"
 import React from "react"
-import { NativeModules, View } from "react-native"
-import { createFragmentContainer, graphql } from "react-relay"
+import { NativeModules, TouchableWithoutFeedback, View } from "react-native"
+import { commitMutation, createFragmentContainer, graphql, RelayProp } from "react-relay"
 import styled from "styled-components/native"
 
 const Constants = NativeModules.ARCocoaConstantsModule
 
 interface ArtworkActionsProps {
   artwork: ArtworkActions_artwork
+  relay?: RelayProp
 }
 
-export const ArtworkActions: React.FC<ArtworkActionsProps> = () => {
-  return (
-    <View>
-      <Flex flexDirection="row">
-        <UtilButton pr={3}>
-          <Box mr={0.5}>
-            <SearchIcon />
-          </Box>
-          <Sans size="3">Save</Sans>
-        </UtilButton>
-        {Constants.AREnabled && (
-          <UtilButton pr={3}>
+export class ArtworkActions extends React.Component<ArtworkActionsProps> {
+  state = {
+    is_saved: null,
```

Comment:
I had trouble following the logic around this flag.

Seems like it's not needed? The optimistic result is used to update the cache, and it gets reverted when the actual result comes in or if an error occurs, so we can use `artwork.is_saved` for everything I think?